rskj-core/src/main/java/co/rsk/net/discovery/PeerExplorer.java

```diff
@@ -242,7 +242,7 @@ private void handlePong(InetSocketAddress pongAddress, PongPeerMessage message)
             this.pendingPingRequests.remove(message.getMessageId());
             NodeChallenge challenge = this.challengeManager.removeChallenge(message.getMessageId());
             if (challenge == null) {
-                this.addConnection(message, request.getAddress().getHostString(), request.getAddress().getPort());
+                this.addConnection(message, request.getAddress().getAddress().getHostAddress(), request.getAddress().getPort());
```

Comment:
By doing this change `Node.getHost()` will store the same value as the one provided by `Node.getAddress().getAddress().getHostAddress()` and we will be loosing the current content of the host var. Instead of update this logic to store a repeated value, wouldn't be better to update the place where the Node.getHost() is used to be be replaced by the current `...getAddress().getHostAddress() `option? 

For me it seems that the problem is where the getHost() is being used instead of the value being provided by node.getHost()

Reply:
the thing is that we do not use any DNS name received from outside. To get an address of a peer that sent us a message in the `handleMessage()` method we use `event.getAddress()`, which returns an instance of `InetSocketAddress` class. In essence, this instance combines two values: actual IP address of a sender and host name resolved locally by reverse dns lookup (see how `InetAddress.getByName()` works). But that host name might make no sense in other network contexts. It can only lead to issues related to specific network setups, cause what we care about is a source IP address of a peer that sends us a message